test/test_macros.hpp

```diff
@@ -11,19 +11,31 @@
 #include <cfloat>
 #include <math.h>
 
-inline float radians(float deg) {
-    return M_PI / 180.0f * deg;
+template<typename Type>
+inline Type radians(Type deg) {
+    return Type(M_PI) / Type(180.0) * deg;
 }
 
-inline float degrees(float rad) {
-    return rad / (M_PI / 180.0f);
+template<typename Type>
+inline Type degrees(Type rad) {
+    return rad / (Type(M_PI) / Type(180.0));
 }
 
 inline bool is_equal(float a, float b, float e) {
     return fabsf(a - b) < e;
 }
+
 inline bool is_equal(float a, float b) {
     return is_equal(a, b, FLT_EPSILON);
 }
 
```

Comment:
I changed them to be templated to get rid of some of the double to float
crap (Wconversion)..

On 16 Aug 2016 21:47, "James Goppert" <notifications@github.com> wrote:

> In test/test_macros.hpp
> <https://github.com/PX4/Matrix/pull/26#discussion_r75006409>:
>
> >  }
> >
> >  inline bool is_equal(float a, float b, float e) {
> >      return fabsf(a - b) < e;
> >  }
> > +
> >  inline bool is_equal(float a, float b) {
> >      return is_equal(a, b, FLT_EPSILON);
> >  }
> >
>
> You should remove any non-template functions, when they get included
> multiple times in different object files it causes issues. Also, I think
> the function I added should work for this.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/PX4/Matrix/pull/26/files/3a1e1e3d2bdfaebd9efd87bdeb9d135dfe6d4453..ee069731db1fe05dfa6e4f1b2999615075e86db7#r75006409>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AERqiab0nZK0gpHyRPMZT7HGj7msDwE-ks5qghPVgaJpZM4JZZwu>
> .
>